### PR TITLE
Fix apicheck CI task and add additional testing for PyPy 3.9

### PR DESCRIPTION
**Pull request by gopackgo90**
Also fixed `apicheck` CI task and added additional testing for PyPy 3.9

### CODE REVIEW

**Review by auvipy — CHANGES_REQUESTED** [review]
can you please check the py3.11 build failure please? there seems to be some distutils import issue

**Comment by gopackgo90** [comment]
Yeah, I was looking at it a bit last night and wasn’t able to figure it out quickly. It’s choking on building the wheels for `python-crontab` and `ephem`. `ephem` has pre-built wheels for Python 3.10, but not 3.11. On my personal macOS I was able to build the wheels for both packages on Python 3.11 without issue, so it will take some more digging. If I can’t figure it out I’ll close this PR and make a new one with the other changes that do work, expanding PyPy/Django matrix testing and fixing the existing Sphinx `apicheck`.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.7', '3.8', '3.9', '3.10', 'pypy-3.9']
+        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11', 'pypy-3.9']
```
Comment (auvipy) [inline .github/workflows/test.yml:15]:
```suggestion
        python-version: ['3.7', '3.8', '3.9', '3.10', 'pypy-3.9']
```

**Review by auvipy — CHANGES_REQUESTED** [review]
for the time being

File: setup.py [hunk 2]
```diff
@@ -39,6 +39,7 @@ def _pyimp():
     Programming Language :: Python :: 3.8
     Programming Language :: Python :: 3.9
     Programming Language :: Python :: 3.10
+    Programming Language :: Python :: 3.11
```
Comment (auvipy) [inline setup.py:42]:
```suggestion
```

**Review by auvipy — CHANGES_REQUESTED** [review]

**Comment by gopackgo90** [comment]
I renamed this PR, I suppose we could just merge those changes in at least as this PR and maybe revisit Python 3.11 later.

**Comment by auvipy** [comment]
yep

**Review by auvipy — APPROVED** [review]

**Pull request merged**